src/blockdata/locktime/relative.rs

```diff
@@ -0,0 +1,289 @@
+    /// assert!(lock.is_satisfied_by_time(Time::from_512_second_intervals(intervals + 10)).expect("a time"));
+    /// ```
+    #[inline]
+    pub fn is_satisfied_by_time(&self, t: Time) -> Result<bool, Error> {
+        use LockTime::*;
+
+        match *self {
+            Time(ref time) => Ok(time.value() <= t.value()),
+            Blocks(ref height) => Err(Error::IncompatibleHeight(*self, *height)),
+        }
+    }
+}
+
+impl From<Height> for LockTime {
+    #[inline]
+    fn from(h: Height) -> Self {
+        LockTime::Blocks(h)
+    }
+}
+
+impl From<Time> for LockTime {
+    #[inline]
+    fn from(t: Time) -> Self {
+        LockTime::Time(t)
+    }
+}
+
+impl fmt::Display for LockTime {
+    #[inline]
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use LockTime::*;
+
+        if f.alternate() {
+            match *self {
+                Blocks(ref h) => write!(f, "block-height {}", h),
+                Time(ref t) => write!(f, "block-time {} (512 second intervals)", t),
+            }
+        } else {
+            match *self {
+                Blocks(ref h) => fmt::Display::fmt(h, f),
+                Time(ref t) => fmt::Display::fmt(t, f),
+            }
+        }
+    }
+}
+
+/// A relative lock time lock-by-blockheight value.
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub struct Height(u16);
+
+impl Height {
+    /// Returns the inner `u16` value.
+    #[inline]
+    pub fn value(self) -> u16 {
+        self.0
+    }
+}
+
+impl From<u16> for Height {
+    #[inline]
+    fn from(value: u16) -> Self {
+        Height(value)
+    }
+}
+
+impl fmt::Display for Height {
+    #[inline]
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Display::fmt(&self.0, f)
+    }
+}
+
+/// A relative lock time lock-by-blocktime value.
+///
+/// For BIP 68 relative lock-by-blocktime locks, time is measure in 512 second intervals.
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub struct Time(u16);
+
+impl Time {
+    /// Create a [`Time`] using time intervals where each interval is equivalent to 512 seconds.
+    ///
+    /// Encoding finer granularity of time for relative lock-times is not supported in Bitcoin.
+    #[inline]
+    pub fn from_512_second_intervals(intervals: u16) -> Self {
+        Time(intervals)
+    }
+
+    /// Create a [`Time`] from seconds, converting the seconds into 512 second interval with ceiling
+    /// division.
+    ///
+    /// # Errors
+    ///
+    /// Will return an error if the input cannot be encoded in 16 bits.
+    #[inline]
+    pub fn from_seconds_ceil(seconds: u32) -> Result<Self, Error> {
```

Comment:
nit: Can this be an `Option<T>` instead? There is only one variant that can happen here. Most arithmetic operations in rust return an Option when the error is only possible before as overflow. 
I would have suggested a dedicated error type, but it would be some work and a lot of code. 

This would be annoying to `unwrap` because we not know all error variants

Reply:
hmmm, a few considerations
1. There is an equivalent method on `Sequence` so we should do the same for both
2. It was a dedicated error type until I added the other variants
3. Option has less information because the error returns the input value, is that useful? I don't know really (@Kixunil is the resident "useful error message" bloke, what do you think man?)

I can do either
1. split the error into two different types
2. Use an `Option` as suggested

I tend to agree that `2` is the best

Reply:
Either way, I'll do it as a follow up please - this one has been running for long enough now.

Reply:
I personally prefer custom error types to `Option` even at the cost of added code. It provides more information when people blindly `?` it. (and `.ok()` is possible if they want to ignore)

A special type would be better indeed.

Reply:
To expand on this a bit, in my experience sprinkling `.ok_or{_else}` or `map_err` everywhere is quite annoying and not doing it will bite you when function deep in the call stack returns error (`None`) and you don't even know which one it was. For instance this is very common when people `?` `std::io::Error`.

Reply:
@sanket1729 could you raise an issue about this, if you feel strongly enough? I am going to merge this PR.

Reply:
Furthering this discussion, I'm interested to learn if our method of having a single enum error type in each module (typically) is the correct way to go about things, alternatives (off the top of my head) are:
- An individual error struct for every error
- An enum for related errors
- An enum error per "some other semantic block" e.g. all of the networking code

Reply:
My initial thoughts: for types convertible from byte arrays and such, parsing errors should be separate from other conversion errors - different layers-different errors
Functions that are very often called together can share error.
Everything else should have its own.

Splitting into crates will force us to at least split errors from independent parts.

Reply:
Don't feel so strongly about it now. My workflow for detecting it is to try using it on a type that I know it should succeed on and match on error types before returning. I will raise issue/PR later about it